### ISSUE TITLE
no-reference suggestion is incomplete?

### ISSUE THREAD

**Issue by bronson** [palantir/tslint#3243]
### Bug Report

- __TSLint version__: 5.7.0
- __TypeScript version__: 2.5.2
- __Running TSLint via__: ts-loader 2.3.7 (Webpack)

#### TypeScript code being linted

tslint says there's a `no-reference` error on the first line of my file:

```ts
///<reference path="png.d.ts"/>
import * as logo from './icon-blue.png'
```

And here's `png.d.ts`:

```ts
declare module '*.png' {
  const value: any
  export = value
}
```

The error is: Disallows /// <reference path=> imports (use ES6-style imports instead). https://palantir.github.io/tslint/rules/no-reference/

However, I don't think I can use es6 imports to import `png.d.ts`.  When I try, Typescript says it isn't a module and doesn't produce any output.

Is there a good way to solve this?

`tslint.json` configuration:

```json
{
  "defaultSeverity": "error",
  "extends": [
    "tslint:recommended",
    "tslint-config-standard"
  ],
  "jsRules": {},
  "rulesDirectory": []
}
```

**Comment by bronson** [comment]
I'm happy to submit a PR to improve the suggestion.  I just don't know what it would be.

**Comment by ajafff** [comment]
Use `import "png";`

If your tsconfig.json includes png.d.ts (via `files` or `include`), you don't need the import at all, because typescript will automatically pick it up.

**Comment by bronson** [comment]
Thank you.  I tried `import 'png'` and `import './png'`...  Same error:

```
ERROR in ./src/ftree.ts
Module not found: Error: Can't resolve 'png' in '/Users/bronson/temproj/src'
 @ ./src/ftree.ts 3:0-13
 @ ./src/content.ts
```

The `png.d.ts` is in `./src`, along with the rest of the source files.  I'm confused why `<reference>` can find the file but `import` can't.

I also tried `import './png.d'`, which found the file but I guess Typescript didn't like that the file only had declarations.  (?)

```
ERROR in ./src/png.d.ts
Module build failed: Error: Typescript emitted no output for /Users/bronson/temproj/src/png.d.ts.
    at successLoader (/Users/bronson/temproj/node_modules/ts-loader/dist/index.js:47:15)
    at Object.loader (/Users/bronson/temproj/node_modules/ts-loader/dist/index.js:29:12)
 @ ./src/ftree.ts 3:0-17
 @ ./src/content.ts
```

**Comment by bronson** [comment]
`"include": ["src/png.d.ts"]` in `tsconfig.json` does work, as you said.  But that's not quite as nice as just importing it from the files that require it.

**Comment by ajafff** [comment]
I actually meant `import "./png";` as you already figured out.
If this doesn't work, it's likely a bug in the loader or maybe even typescript.
There's nothing we can do to fix this on the TSLint side.

**Comment by bronson** [comment]
You're right, I'm pretty sure it's a bug in the loader.  Sorry for the false implication, I'll file a bug over there.

Gotta say, it's a big relief that this is supposed to work.  The world is saner than I thought.  :)
